pkg/helpers/crd.go

```diff
@@ -51,16 +52,21 @@ func CreateGitRepo(obj metav1.Object, gvk schema.GroupVersionKind, template *syn
 
 		err = client.Get(context.TODO(), namespacedName, existingRepo)
 		if err != nil {
-			return false, fmt.Errorf("could not update existing repo: %v", err)
+			return fmt.Errorf("could not update existing repo: %v", err)
 		}
 
-		existingRepo.Spec = repo.Spec
+		if !reflect.DeepEqual(existingRepo.Spec, repo.Spec) {
+			existingRepo.Spec = repo.Spec
+
+			err = client.Update(context.TODO(), existingRepo)
+		}
+
+		return err
 
-		return false, client.Update(context.TODO(), existingRepo)
 	} else if err != nil {
-		return false, err
+		return err
 	}
-	return true, nil
+	return nil
```

Comment:
You can collapse all the preceding returns into one `return err` and also remove the `else if`